### PR TITLE
doc: front page of etcd3 doc

### PR DESCRIPTION
**Pull request by xiang90**
/cc @heyitsanthony @joshix @weiliendang

Fix https://github.com/coreos/etcd/issues/5106

### CODE REVIEW

File: Documentation/docs.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+ - Why etcd
+ - Concepts
+ - Glossary
+
+## Upgrading and Compatibility
+
+## Troubleshooting
+
+[download_build]: ./dl_build.md
```
Comment (joshix) [inline Documentation/docs.md]: just `dl_build.md` will do for the path

File: Documentation/docs.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
+# Documentation
+
+etcd is a distributed key-value store designed for serving most critical data. It enables reliable distributed coordination, like distributed locking, leader election and barrier. Once an etcd cluster is set up, it is supposed to be there for you "forever".
```
Comment (joshix) [inline Documentation/docs.md]: etcd is a distributed key-value store designed to reliably and quickly preserve and provide access to critical data. It enables reliable distributed coordination of this data, through distributed locking, leader elections, and write barriers. An etcd cluster is intended for high availability and permanent data storage and retrieval.

**Comment by heyitsanthony** [comment]
via style linter
```
docs.md.3: eschew you ("etcd is a distributed key-value store designed for serving most critical data. It enables reliable distributed coordination, like distributed locking, leader election and barrier. Once an etcd cluster is set up, it is supposed to be there for you \"forever\".")
docs.md.5: header: word capitalized ("Started" in "## Getting Started")
docs.md.37: header: word capitalized ("Compatibility" in "## Upgrading and Compatibility")
```

**Comment by xiang90** [comment]
@joshix Fixed. PTAL.

**Comment by xiang90** [comment]
@heyitsanthony Good job...

**Comment by xiang90** [comment]
@heyitsanthony Fixed.

File: Documentation/docs.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+ - Configuration
+ - Security
+ - Monitoring
+ - Maintenance
+ - Disaster recovery
+ - Performance
+
+## Learning
+
+Want to learn more about the concept and internal about etcd? Read the following pages!
```
Comment (heyitsanthony) [inline Documentation/docs.md]: s/concept and internal about/concepts and internals behind/

**Comment by heyitsanthony** [comment]
lgtm aside from the concepts/internals nit

**Pull request merged**